### PR TITLE
Curve security support. Added/fixed some tests

### PR DESCRIPTION
**Pull request by tbibby**
Added the ability to add curve security to a socket.
Some tests were not passing - fixed.
Added some new tests with different ZeroMQ patterns.
Please shout if you think anything needs more work!

### CODE REVIEW

File: Tests/SocketOptionsTest.swift [hunk 1]
```diff
@@ -38,66 +38,67 @@ class SocketOptionsTest: XCTestCase {
...
         XCTAssertEqual(try socket.getReconnectInterval(), 9)
-
+        
         try socket.setMaxReconnectInterval(10)
         XCTAssertEqual(try socket.getMaxReconnectInterval(), 10)
-
+        
         try socket.setBacklog(11)
         XCTAssertEqual(try socket.getBacklog(), 11)
     }
```
Comment (azawawi) [inline Tests/SocketOptionsTest.swift]: @tbibby Can you please remove unnecessary white space indentation from this change.

**Comment by azawawi** [comment]
@tbibby Fantastic. Thanks for your contribution. I will merge it once you remove the extra white space.

**Comment by tbibby** [comment]
Thanks! I have removed the extra white space. The watch builds still aren't succeeding in Travis-CI due to an incorrect simulator specification - do you know what I need to do to make them pass?

**Review by azawawi — APPROVED** [review]
Code review. done. Thanks.

**Comment by azawawi** [comment]
I will fix them now. Thanks for your pull request 👍

**Pull request merged**